### ISSUE TITLE
Calling Contacts.getAll on Android - gets the contacts but not the phone number or the emails

### ISSUE THREAD

**Issue by dovfrank** [rt2zz/react-native-contacts#44]
Works fine on iOS.

**Comment by morenoh149** [comment]
so you get the contacts. But the objects lack the number and email fields?

please share example output and sample code

**Comment by dovfrank** [comment]
Usage example:

```
 let rn_contacts = require('react-native-contacts');
 rn_contacts.getAll((err, contacts) => {
       console.log(contacts)
 })
```
I took a screenshot of the output:
![bug in contacts](https://cloud.githubusercontent.com/assets/1114736/14065083/72eabde0-f423-11e5-9219-2553f5c89097.png)

You can see that I got the contact name and phone but the phone numbers and emails are empty.
Not sure if it helps but I'm using HTC M8.

**Comment by morenoh149** [comment]
Rn version and module version? 

If you're on the latest module version try a version 2 releases ago. We recently changed the table lookup on Android which may be the reason you're seeing that.

**Comment by dovfrank** [comment]
RN 0.21.0
Module 0.2.1

**Comment by morenoh149** [comment]
That's odd. The major change was for 0.2.2 -> 0.2.3 try the latest then. Also what do you see in an android simulator?

**Comment by dovfrank** [comment]
Sorry man, my mistake. I was on "^0.2.1", when changing to "0.2.1" it works fine again.
Thanks!

**Comment by dovfrank** [comment]
Hi,
After changing to Module number 0.2.1, it works on Android but not on iPhone.
It crashes with exception:
> Exception '*** setObjectForKey: object cannot be nil (key: thumbnailPath)' was thrown while invoking getAll on target RCTContacts with params (
    475
)

I guess that the error is on:
 `[contact setObject: [self getABPersonThumbnailFilepath:person] forKey:@"thumbnailPath"];`

Again, this happens when I call
` rn_contacts.getAll((err, contacts) => {`

**Comment by morenoh149** [comment]
seems related to https://github.com/rt2zz/react-native-contacts/commit/004909f569545fc57a0c1c6237b4f8bd9a38d886 try 0.2.0 and report back

**Comment by dovfrank** [comment]
Tried 0.2.0 but the packager fails to run.
Giving me the following error:

> Error: Naming collision detected: /Users/MyApp/node_modules/react-native-contacts/test/node_modules/react-native/Libraries/vendor/react/browser/eventPlugins/TouchHistoryMath.js collides with /Users/MyApp/node_modules/react-native/Libraries/vendor/react/browser/eventPlugins/TouchHistoryMath.js
    at HasteMap._updateHasteMap (HasteMap.js:132:13)
    at HasteMap.js:103:28
    at tryCallOne (/Users/MyApp/node_modules/promise/lib/core.js:37:12)
    at /Users/MyApp/node_modules/promise/lib/core.js:123:15
    at flush (/Users/MyApp/node_modules/asap/raw.js:50:29)
    at nextTickCallbackWith0Args (node.js:453:9)
    at process._tickCallback (node.js:382:13)

**Comment by morenoh149** [comment]
doesn't seem related. `/Users/MyApp/node_modules/react-native-contacts/test/node_modules/react-native/Libraries/vendor/react/browser/eventPlugins/TouchHistoryMath.js` moreover `react-native-contacts` doesn't have a `test` dir

**Comment by dovfrank** [comment]
Tried again, this is what I got:

> Error: Naming collision detected: /ReactNative/MyApp/node_modules/react-native/packager/react-packager/src/Resolver/polyfills/String.prototype.es6.js collides with /ReactNative/MyApp/node_modules/react-native-contacts/test/node_modules/react-native/packager/react-packager/src/Resolver/polyfills/String.prototype.es6.js
    at HasteMap._updateHasteMap (HasteMap.js:132:13)
    at HasteMap.js:103:28
    at tryCallOne (/ReactNative/MyApp/node_modules/promise/lib/core.js:37:12)
    at /ReactNative/MyApp/node_modules/promise/lib/core.js:123:15
    at flush (/ReactNative/MyApp/node_modules/asap/raw.js:50:29)
    at nextTickCallbackWith0Args (node.js:453:9)
    at process._tickCallback (node.js:382:13)

Note that test folder is created and is there.
![screen shot 2016-04-03 at 11 02 24 am](https://cloud.githubusercontent.com/assets/1114736/14231434/b65374e2-f98b-11e5-8b64-c100882ad1b9.png)

**Comment by morenoh149** [comment]
ah yeah 0.2.0 accidentally had a tests dir https://github.com/rt2zz/react-native-contacts/commit/544a8f9ca3ff7654b5aeeffbbbb309afd6081d3b try one earlier than that :p

**Comment by dovfrank** [comment]
Tried on 0.1.5
Does not work on iPhone:

> ExceptionsManager.js:76 Exception '*** setObjectForKey: object cannot be nil (key: thumbnailPath)' was thrown while invoking getAll on target RCTContacts with params (
    351
)

**Comment by morenoh149** [comment]
that's a separate issue lol. I just published 0.2.4 which should address the original bug you saw. Please confirm it works.

**Comment by dovfrank** [comment]
Seems to work now.
Thanks!
